test/unit/test_training.py

```diff
@@ -83,73 +84,58 @@ def test_single_machine(run_module, single_machine_training_env):
                                   single_machine_training_env.to_env_vars())
 
 
-@pytest.mark.skipif(sys.version_info.major != 3,
```

Comment:
This tests are flaky on py2 because of the dict key ordering. I think we should add the skipif back.